.github/workflows/macos-integration-tests.yml

```
@@ -0,0 +1,224 @@
+          brew install unixodbc autoconf automake gflags postgresql zlib
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_DEFAULT_REGION }}
+
+      - name: Set up Temp AWS Credentials
+        run: |
+          creds=($(aws sts get-session-token \
+            --duration-seconds 21600 \
+            --query 'Credentials.[AccessKeyId, SecretAccessKey, SessionToken]' \
+            --output text \
+          | xargs));
+          echo "::add-mask::${creds[0]}"
+          echo "::add-mask::${creds[1]}"
+          echo "::add-mask::${creds[2]}"
+          echo "AWS_ACCESS_KEY_ID=${creds[0]}" >> $GITHUB_ENV
+          echo "AWS_SECRET_ACCESS_KEY=${creds[1]}" >> $GITHUB_ENV
+          echo "AWS_SESSION_TOKEN=${creds[2]}" >> $GITHUB_ENV
+
+      - name: Setup Cluster Id and Shard Id Env Variables
+        run: |
+          echo "AURORA_CLUSTER_ID=AWS-PGODBC-MacOS-Integ-${{github.run_id}}" >> $GITHUB_ENV
+
+      - name: Create Aurora Cluster
+        id: auroraClusterSetup
+        shell: bash
+        working-directory: macos
+        run: |
+          . "./aws_rds_helper"
+          create_aurora_rds_cluster \
+            ${{ secrets.TEST_USERNAME }} \
+            ${{ secrets.TEST_PASSWORD }} \
+            ${{ env.TEST_DATABASE }} \
+            ${{ env.AURORA_CLUSTER_ID }} \
+            ${{ env.NUM_AURORA_INSTANCES }} \
+            ${{ env.ENGINE }} \
+            ${{ env.ENGINE_VERSION }} \
+            ${{ secrets.AWS_DEFAULT_REGION }}
+
+      - name: Get Aurora Cluster endpoint
+        shell: bash
+        working-directory: macos
+        run: |
+          . "./aws_rds_helper"
+          endpoint=$(get_cluster_endpoint ${{ env.AURORA_CLUSTER_ID }})
+          echo "AURORA_CLUSTER_ENDPOINT=$endpoint" >> $GITHUB_ENV
+        env:
+          AWS_ACCESS_KEY_ID: ${{ env.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ env.AWS_SECRET_ACCESS_KEY }}
+          AWS_SESSION_TOKEN: ${{ env.AWS_SESSION_TOKEN }}
+
+      - name: Create Aurora Cluster Secrets
+        shell: bash
+        working-directory: macos
+        run: |
+            . "./aws_rds_helper"
+            secretsArn=$(create_db_secrets \
+                ${{ secrets.TEST_USERNAME }} \
+                ${{ secrets.TEST_PASSWORD }} \
+                ${{ env.ENGINE }} \
+                ${{ env.AURORA_CLUSTER_ENDPOINT }})
+            echo "::add-mask::$secretsArn"
+            echo "AURORA_CLUSTER_SECRETS_ARN=$secretsArn" >> $GITHUB_ENV
+
+      - name: Cache AWS SDK libraries
+        id: cache-static-aws-sdk
+        uses: actions/cache@v4
+        if: always() && steps.cache-static-aws-sdk.outputs.cache-hit != 'true'
+        with:
+          path: ./libs/aws-rds-odbc/aws_sdk
+          key: ${{ runner.os }}-aws-sdk-cache-lib
```

Reuse the cache from the regression tests. We don't need to store 2 versions of the MacOS AWS SDK
i.e. use the same path & key, case sensitive
```suggestion
      - name: Cache AWS SDK libraries
        id: cache-static-aws-sdk
        uses: actions/cache@v4
        if: always() && steps.cache-static-aws-sdk.outputs.cache-hit != 'true'
        with:
          path: |
            libs/aws-rds-odbc/aws_sdk
          key: ${{ runner.os }}-aws-sdk-dynamic-lib
```